spec.md

```diff
@@ -0,0 +1,954 @@
+}
+```
+
+##### ShutDownMachine Errors
+
+If the plugin is unable to complete the GetMachine call successfully, it MUST return a non-ok gRPC code in the gRPC status.
+If the conditions defined below are encountered, the plugin MUST return the specified gRPC error code.
+
+| gRPC Code | Condition | Description | Recovery Behavior | Auto Retry Required |
+|-----------|-----------|-------------|-------------------|------------|
+| 0 OK | Successful | The termination request for the VM was accepted (or) the VM is already in terminated state. |  | N |
+| 1 CANCELED | Cancelled | Call was cancelled. Perform any pending clean-up tasks and return the call |  | N |
+| 2 UNKNOWN | Something went wrong | Not enough information on what went wrong | Retry operation after sometime | Y |
+| 3 INVALID_ARGUMENT | Re-check supplied parameters | Re-check the supplied `MachineID` and make sure that it is in the desired format and not a blank value. Exact issue to be given in `.message` | Update `MachineID` to fix issues. | N |
+| 4 DEADLINE_EXCEEDED | Timeout | The call processing exceeded supplied deadline | Retry operation after sometime | Y |
+| 5 NOT_FOUND |  |  |  |  |
+| 6 ALREADY_EXISTS |  |  |  |  |
+| 7 PERMISSION_DENIED | Insufficent permissions | The requestor doesn't have enough permissions to shut down a VM and it's required dependencies | Update requestor permissions to grant the same | N |
+| 8 RESOURCE_EXHAUSTED |  |  |  |  |
+| 9 PRECONDITION_FAILED | VM is in inconsistent state | The VM is in a state that is invalid for this operation | Manual intervention might be needed to fix the state of the VM | N |
+| 10 ABORTED |  |  |  |  |
+| 11 OUT_OF_RANGE |  |  |  |  |  |
+| 12 UNIMPLEMENTED | Not implemented | Unimplemented indicates operation is not implemented or not supported/enabled in this service. | Retry with an alternate logic or implement this method at the plugin. Most methods by default are in this state | N |
+| 13 INTERNAL | Major error | Means some invariants expected by underlying system has been broken. If you see one of these errors, something is very broken. | Needs manual intervension to fix this | N |
+| 14 UNAVAILABLE | Not Available | Unavailable indicates the service is currently unavailable. | Retry operation after sometime | Y |
+| 15 DATALOSS |  |  |  |  |
+| 16 UNAUTHENTICATED | Missing provider credentials | Request does not have valid authentication credentials for the operation | Fix the provider credentials | N |
+
+The status `message` MUST contain a human readable description of error, if the status `code` is not `OK`.
+This string MAY be surfaced by CMI-Client to end users.
+
+#### `ListMachines`
+
+A Plugin MUST implement this RPC call if it has `LIST_MACHINES` capability.
+The Plugin SHALL return the information about all the machines associated with the `ProviderSpec`.
+The CMI-Client SHALL NOT expect a consistent "view" of all machines when paging through the machines list via multiple calls to `ListMachines`.
```

Comment:
You mention "paging" here but don't actually describe a paging mechanism. Do you intend for the plugin to page responses, similarly to CSI spec? If yes, this should be described in more details.
I personally don't like pages since it makes things more complex for everybody, so if you think it can be avoided, by all means let's do without it. I am not sure however if that would work for all cases and providers, you might run into some limits related to the size of responses for very large sets.